README.md

```diff
@@ -7,12 +7,14 @@
 Nix modules for darwin, `/etc/nixos/configuration.nix` for macOS.
 
 This project aims to bring the convenience of a declarative system approach to macOS.
-Nix-darwin is built up around [Nixpkgs](https://github.com/NixOS/nixpkgs), quite similar to [NixOS](https://nixos.org/).
+nix-darwin is built up around [Nixpkgs](https://github.com/NixOS/nixpkgs), quite similar to [NixOS](https://nixos.org/).
 
 ## Install
 
 To install nix-darwin, a working installation of [Nix](https://github.com/NixOS/nix#installation) is required.
 
+> NOTE: Using `darwin-installer` is no longer necessary on flake based systems.
```

Comment:
So far, the only thing I've moved to regular activation is creating `/run`.

The things I didn't move to regular activation:

- Copying example configuration to `~/.nixpkgs/darwin-configuration.nix`
- Sourcing the static versions of `/etc/bashrc` and `/etc/zshrc` as `programs.bash.enable` defaults to `true` and `programs.zsh.enable = true;` is uncommented in both `simple.nix` and the example `flake.nix`
- Adding `darwin` to `nix-channel`
- Setting `nix.configureBuildUsers = true;` to migrate old style users to new style users, as there's a check that ensures it's set when old style users are present.

Reply:
This used to be more but more modules have been added since the initial version so that's great.

I think the main functional change in that case is the fact that the known hashes for zshrc/bashrc will always need to be correct since there's no installer that edits the existing files anymore. In that case perhaps it's best to also remove those steps from the installer so that both installation paths work the same.

Reply:
I've removed the logic in `darwin-installer` to update `/etc/bashrc` and `/etc/zshrc`, as well as removed the `nix.configureBuildUsers = true;`.

Reply:
@LnL7 this PR is ready to review now :smile: 

I've updated the known hashes for `/etc/nix/nix.conf` and `/etc/{bash,zsh}rc`, I've also added a copy of known files as reference for future.

I've also changed the `etc` activation script to fail if any of the files don't match the known hash and need to be replaced manually.